diff --git a/modules/simpletest/drupal_web_test_case.php b/modules/simpletest/drupal_web_test_case.php
index 6d0e59a..d0bf6e4 100644
--- a/modules/simpletest/drupal_web_test_case.php
+++ b/modules/simpletest/drupal_web_test_case.php
@@ -2357,13 +2357,13 @@ class DrupalWebTestCase extends DrupalTestCase {
    */
   protected function handleForm(&$post, &$edit, &$upload, $submit, $form) {
     // Retrieve the form elements.
-    $elements = $form->xpath('.//input[not(@disabled)]|.//textarea[not(@disabled)]|.//select[not(@disabled)]');
+    $elements = $form->xpath('.//input[not(@disabled)]|.//button[not(@disabled)]|.//textarea[not(@disabled)]|.//select[not(@disabled)]');
     $submit_matches = FALSE;
     foreach ($elements as $element) {
       // SimpleXML objects need string casting all the time.
       $name = (string) $element['name'];
-      // This can either be the type of <input> or the name of the tag itself
-      // for <select> or <textarea>.
+      // This can either be the type of <input> or <button> or the name of the
+      // tag itself for <select> or <textarea>.
       $type = isset($element['type']) ? (string) $element['type'] : $element->getName();
       $value = isset($element['value']) ? (string) $element['value'] : '';
       $done = FALSE;
